### ISSUE TITLE
How are cells numbered in messages?

### ISSUE THREAD

**Issue by wstomv** [nbQA-dev/nbQA#529]
When I run `nbqa` on a notebook with `flake8`, `mypy`, or `pylint`, I see messages with the following structure
```
    <file_name>:cell_#:...
```
How is that cell number determined?

I have not been able to locate the indicated cells.

When using a tool like _NbToolbelt_ ([see PyPI](https://pypi.org/project/nbtoolbelt/)), with the command `nbtb dump`, you get a listing of the notebook with _all_ the cells numbered starting from 0 (the index in the JSON array of all cells). It is also not the number when assuming numbering starts at 1.

It also appears not to be the 0-based or 1-based number of just the code cells (the execution count).

Maybe I am overlooking the obvious. I did not find this explained in the documentation.

**Comment by MarcoGorelli** [comment]
Hi @wstomv 

Thanks for your report - it should just be the 1-based number of code cells, in the order in which they appear. Is that not what you're seeing? If you could share an example that'd be really useful

This should be documented anyway, so thanks for highlighting this area for improvement

**Comment by jukworks** [comment]
Although cell_# gives me the precise error position, how do you guys find/locate the cell_#? It's difficult to find cell_39 quickly.
I couldn't see any options like "show cell number" in Jupyterlab. Any hints?

**Comment by MarcoGorelli** [comment]
> It's difficult to find cell_39 quickly.

Yeah, true. If you execute it top-to-bottom, then you could look at the execution count

I did consider using the execution count in the output, but then there could be duplicates so it'd be even more confusing. Am open to suggestions anyway

**Comment by s-weigand** [comment]
Another problem with execution count is that it becomes useless when you use tools like [nbstripout](https://github.com/kynan/nbstripout) since it deletes the output and execution count with default settings.

**Comment by s-weigand** [comment]
> It's difficult to find cell_39 quickly.

As a quick and dirty workaround for `jupyterlab` you can open the devTools of your browser, go to the console tab and paste this code snipped (when I got some time I might write a proper extension).
```js
function find_code_cell(cell_nr) {
  var cell = document.querySelectorAll(".jp-CodeCell")[cell_nr];
  Array.from(document.querySelectorAll(".jp-mod-active")).map((activeCell) => {
    activeCell.classList.remove("jp-mod-active");
  });
  cell.scrollIntoView();
  cell.classList.add("jp-mod-active");
}
```
After that, you can use the console to find cells.
E.g.
```js
find_code_cell(10)
```

**Comment by MarcoGorelli** [comment]
nice, thanks @s-weigand !

**Comment by MarcoGorelli** [comment]
closed in e224feb
